### ISSUE TITLE
fix: typescript error TS2344 in fast-foundation.d.ts with TypeScript 4.8.2

### ISSUE THREAD

**Issue by jedwards1211** [microsoft/fast#6365]
# 🐛 Bug Report

Starting with TypeScript 4.8.2 there are errors like

```
node_modules/.pnpm/@microsoft+fast-foundation@2.46.14/node_modules/@microsoft/fast-foundation/dist/fast-foundation.d.ts:2252:66 - error TS2344: Type 'T' does not satisfy the constraint 'string | number | boolean | symbol | any[] | Uint8Array | ({ createCSS?(): string; } & Record<PropertyKey, any>) | null'.
  Type 'T' is not assignable to type '{ createCSS?(): string; } & Record<PropertyKey, any>'.
    Type 'T' is not assignable to type '{ createCSS?(): string; }'.

2252 declare function create<T>(nameOrConfig: string): CSSDesignToken<T>;

  node_modules/.pnpm/@microsoft+fast-foundation@2.46.14/node_modules/@microsoft/fast-foundation/dist/fast-foundation.d.ts:2252:25
    2252 declare function create<T>(nameOrConfig: string): CSSDesignToken<T>;
                                 ~
    This type parameter might need an `extends { createCSS?(): string; }` constraint.
  node_modules/.pnpm/@microsoft+fast-foundation@2.46.14/node_modules/@microsoft/fast-foundation/dist/fast-foundation.d.ts:2252:25
    2252 declare function create<T>(nameOrConfig: string): CSSDesignToken<T>;
                                 ~
    This type parameter might need an `extends { createCSS?(): string; } & Record<PropertyKey, any>` constraint.
  node_modules/.pnpm/@microsoft+fast-foundation@2.46.14/node_modules/@microsoft/fast-foundation/dist/fast-foundation.d.ts:2252:25
    2252 declare function create<T>(nameOrConfig: string): CSSDesignToken<T>;
                                 ~
    This type parameter might need an `extends string | number | boolean | symbol | any[] | Uint8Array | ({ createCSS?(): string; } & Record<PropertyKey, any>) | null` constraint.
```

`CSSDesignToken` is declared as
```ts
export declare interface CSSDesignToken<T extends string | number | boolean | BigInteger | null | Array<any> | symbol | ({
    createCSS?(): string;
} & Record<PropertyKey, any>)> extends DesignToken<T>, CSSDirective {
```

## 💻 Repro or Code Sample

[TS Playground Example of the error](https://www.typescriptlang.org/play?ts=4.8.2#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXzDigxAB4AVAPgApUoBbEAeRgGE9EsBzALngGcMMLKi4BKPqwDKUgCIh+3VORwBrEKgqUA3AChdIAB4AHHDAzxQkWAhEkYiKGATS5CpSvWby8IyVTA-AJCIlzwAD7wqMj0AEYgMBHwsTg4ECBQ+JEAQtwAkqgkXAlJ0RAQSQCCMDBQAJ6kmXWUSfx1cWlJ1ADeuvD9BEQkrgD81BLBwqJ6AL7wAGTwAErgZsCkAAowOMYJGHUA0iB1ADTwTZRilH4agfDyilzKahpa-b26MwYmZhZW0HB4HYEo5nPd3E9PK8fDcAkFBFMwpFonESpEUmkMll4LkuAUimioshylUavVGqhmq12ikKpFujMWr0ZkA)

## 🌍 Your Environment

TypeScript 4.8.2

**Comment by TrevorKarjanis** [comment]
I experienced the same with TS 4.6.4. [skipLibCheck](https://www.typescriptlang.org/tsconfig#skipLibCheck) resolves the issue if necessary.

**Comment by chrisdholt** [comment]
Thanks for confirming the issue beginning in 4.8!

**Comment by rajsite** [comment]
@jedwards1211 this issue should now be resolved with merging of #6779 and release `@microsoft/fast-foundation@2.49.2`

**Comment by rajsite** [comment]
@chrisdholt we've been having good success with the changes from #6779 so I think that resolves this issue, i.e. fixed in `@microsoft/fast-foundation@2.49.2`

**Comment by janechu** [comment]
Unfortunately `@microsoft/fast-foundation` is being deprecated, refer to #6955.
